perf_bench_test.go

```diff
@@ -24,127 +24,567 @@ package main
 )
 
 var (
-	isColdStart     = flag.Bool("coldStart", false, "Profile cold starts (default is false)")
-	vmNum           = flag.Int("vm", 2, "The number of VMs")
-	targetReqPerSec = flag.Int("requestPerSec", 2, "The target number of requests per second")
-	executionTime   = flag.Int("executionTime", 2, "The execution time of the benchmark in seconds")
+	// shared arguments
+	warmUpTime   = flag.Float64("warmUpTime", 5, "The warm up time before profiling in seconds")
+	coolDownTime = flag.Float64("coolDownTime", 1, "The cool down time after profiling in seconds")
+	profileTime  = flag.Float64("profileTime", 10, "The profiling time in seconds")
+	funcNames    = flag.String("funcNames", "helloworld", "Names of the functions to benchmark, separated by comma")
+	isBindSocket = flag.Bool("bindSocket", false, "Bind all VMs to socket 1")
+
+	vmNum     = flag.Int("vm", 2, "TestProfileSingleConfiguration: The number of VMs")
+	targetRPS = flag.Int("rps", 10, "TestProfileSingleConfiguration: The target requests per second")
+
+	vmIncrStep = flag.Int("vmIncrStep", 4, "TestProfileIncrementConfiguration: The increment VM number")
+	maxVMNum   = flag.Int("maxVMNum", 100, "TestProfileIncrementConfiguration: The maximum VM number")
+
+	// profiler arguments
+	profilerLevel    = flag.Int("l", 1, "Profile level (-l flag)")
+	profilerInterval = flag.Uint64("I", 500, "Print count deltas every N milliseconds (-I flag)")
+	profilerNodes    = flag.String("nodes", "", "Include or exclude nodes (with "+
+		"+ to add, "+
+		"-|^ to remove, "+
+		"comma separated list, wildcards allowed, "+
+		"add * to include all children/siblings, "+
+		"add /level to specify highest level node to match, "+
+		"add ^ to match related siblings and metrics, "+
+		"start with ! to only include specified nodes)")
 )
 
-func TestBenchRequestPerSecond(t *testing.T) {
+func TestProfileIncrementConfiguration(t *testing.T) {
+	var (
+		idx, rps      int
+		pinnedFuncNum int
+		startVMID     int
+		servedTh      uint64
+		isSyncOffload bool = true
+		metrFile           = "benchRPS.csv"
+		images             = getImages(t)
+		cores              = runtime.NumCPU()
+		metrics            = make([]map[string]float64, *maxVMNum / *vmIncrStep)
+	)
+	log.SetLevel(log.InfoLevel)
+
+	checkInputValidation(t)
+
+	createResultsDir()
+
+	funcPool = NewFuncPool(!isSaveMemoryConst, servedTh, pinnedFuncNum, isTestModeConst)
+
+	plrfncPid := pullImages(t, images)
+	for vmNum := *vmIncrStep; vmNum <= *maxVMNum; vmNum += *vmIncrStep {
+		if vmNum < cores {
+			rps = calculateRPS(vmNum)
+		} else {
+			rps = calculateRPS(cores)
+		}
+
+		log.Infof("vmNum: %d, Target RPS: %d", vmNum, rps)
+
+		bootVMs(t, images, startVMID, vmNum, plrfncPid)
+		metrics[idx] = loadAndProfile(t, images, vmNum, rps, isSyncOffload)
+		startVMID = vmNum
+		idx++
+	}
+
+	dumpMetrics(t, metrics, metrFile)
+	profile.CSVPlotter(*vmIncrStep, *benchDir, metrFile)
+
+	tearDownVMs(t, images, startVMID, isSyncOffload)
+}
+
```

Comment:
please add a comment, describing what kind of experiment this test runs

Reply:
Done